test/views/OptionalSupportingDocsViewSpec.scala

```diff
@@ -0,0 +1,122 @@
+import org.jsoup.nodes.Document
+import play.api.data.Form
+import play.api.mvc.Call
+import play.twirl.api.Html
+import views.html.OptionalSupportingDocsView
+
+class OptionalSupportingDocsViewSpec extends ViewBaseSpec with BaseMessages {
+
+  private lazy val injectedView: OptionalSupportingDocsView = app.injector.instanceOf[OptionalSupportingDocsView]
+
+  val formProvider: OptionalSupportingDocsFormProvider = injector.instanceOf[OptionalSupportingDocsFormProvider]
+
+  val backLink: Call = controllers.routes.AnyOtherSupportingDocsController.onLoad()
+
+  "Rendering the Optional Supporting Documents page" when {
+
+    "no errors exist" should {
+
+      val form: Form[Seq[String]] = formProvider.apply()
+      lazy val view: Html = injectedView(form, backLink, Seq.empty)(fakeRequest, messages)
+      lazy implicit val document: Document = Jsoup.parse(view.body)
+
+      s"have the correct page title" in {
+        document.title mustBe OptionalDocumentsMessages.pageTitle
+      }
+
+      s"have the correct page header" in {
+        document.title mustBe OptionalDocumentsMessages.pageHeader
+      }
+
+      "not render an error summary" in {
+        document.select("div.govuk-error-summary").size mustBe 0
+      }
+
+      "not render an error message against the field" in {
+        document.select("#value-error").size mustBe 0
+      }
+    }
+
+    "error exist when no option selected exist" should {
+
+      val form: Form[Seq[String]] = formProvider().bind(Map("" -> ""))
+      lazy val view: Html = injectedView(form, backLink, Seq.empty)(fakeRequest, messages)
+      lazy implicit val document: Document = Jsoup.parse(view.body)
+
+      s"have the correct page title" in {
+        document.title mustBe errorPrefix + OptionalDocumentsMessages.pageTitle
+      }
+
+      s"have the correct page header" in {
+        document.title mustBe errorPrefix + OptionalDocumentsMessages.pageHeader
+      }
+
+      "render an error summary with the correct message" in {
+        elementText("div.govuk-error-summary > div") mustBe OptionalDocumentsMessages.errorRequired
+      }
+
+      "render an error message against the field" in {
+        elementText("#optionalDocumentsList-error") mustBe errorPrefix + OptionalDocumentsMessages.errorRequired
+      }
+    }
+
+  }
+
+  it should {
+
+    val form: Form[Seq[String]] = formProvider.apply()
+    lazy val view: Html = injectedView(form, backLink, Seq.empty)(fakeRequest, messages)
+    lazy implicit val document: Document = Jsoup.parse(view.body)
+
+    s"have the correct hint" in {
+      elementText("#optionalDocumentsList-hint") mustBe OptionalDocumentsMessages.hint
+    }
+    "render a back link with the correct URL" in {
+      elementAttributes("#back-link") must contain("href" -> backLink.url)
+    }
+
+    s"the checkbox for importAndEntry is rendered" in {
+      document.getElementsByAttributeValue("for", "importAndEntry").get(0).text() mustBe OptionalDocumentsMessages.importAndEntry
```

Comment:
Just select the single element that has the correct attribute for the given class (see below):

elementText(".govuk-checkboxes__label[for=importAndEntry]") ...